core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1003,82 +1061,113 @@ public void week() {
 
     when(nullRef.type()).thenReturn(INTEGER);
     when(missingRef.type()).thenReturn(INTEGER);
-    assertEquals(nullValue(), eval(DSL.week(DSL.literal("2019-01-05"), nullRef)));
-    assertEquals(missingValue(), eval(DSL.week(DSL.literal("2019-01-05"), missingRef)));
+    assertEquals(nullValue(), eval(DSL.week(
+        functionProperties,
+        DSL.literal("2019-01-05"), nullRef)));
+    assertEquals(missingValue(), eval(DSL.week(
+        functionProperties,
+        DSL.literal("2019-01-05"), missingRef)));
 
     when(nullRef.type()).thenReturn(DATE);
     when(missingRef.type()).thenReturn(INTEGER);
-    assertEquals(missingValue(), eval(DSL.week(nullRef, missingRef)));
+    assertEquals(missingValue(), eval(DSL.week(
+        functionProperties,
+        nullRef, missingRef)));
 
     FunctionExpression expression = DSL
-        .week(DSL.literal(new ExprTimestampValue("2019-01-05 01:02:03")));
+        .week(functionProperties, DSL.literal(new ExprTimestampValue("2019-01-05 01:02:03")));
     assertEquals(INTEGER, expression.type());
     assertEquals("week(TIMESTAMP '2019-01-05 01:02:03')", expression.toString());
     assertEquals(integerValue(0), eval(expression));
 
-    expression = DSL.week(DSL.literal("2019-01-05"));
+    expression = DSL.week(functionProperties, DSL.literal("2019-01-05"));
     assertEquals(INTEGER, expression.type());
     assertEquals("week(\"2019-01-05\")", expression.toString());
     assertEquals(integerValue(0), eval(expression));
 
-    expression = DSL.week(DSL.literal("2019-01-05 00:01:00"));
+    expression = DSL.week(functionProperties, DSL.literal("2019-01-05 00:01:00"));
```

Comment:
Is it possible to split tests or use `assertAll`?

Reply:
Refactored and added tests in https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/3c843ba8cff46cca9ff731b204245ef8adab58bc, https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/c664af7ba579b2eba3147ff9b05e2a6971ae8cd1, and https://github.com/Bit-Quill/opensearch-project-sql/pull/199/commits/dcbe2e22598d5404f4dee2e606ce5228aa1b7a9f